csharp/src/Ice/LegacyTransceiver.cs

```diff
@@ -87,6 +88,14 @@ public override async ValueTask<TransceiverStream> AcceptStreamAsync(Cancellatio
                     await ReceiveAsync(buffer.Slice(Ice1Definitions.HeaderSize), cancel).ConfigureAwait(false);
                 }
 
+                // Make sure the transceiver is marked as validated. This flag is necessary because incoming
+                // connection initialization doesn't wait for connection validation message. So the connection
+                // is considered validated on the server side only once the first frame is received. This is
+                // only useful for connection warnings, to prevent a warning from showing up if the server side
+                // connection is closed before the first message is received (which can occur with SSL for
+                // example if the certification validation fails on the client side).
+                IsValidated = true;
```

Comment:
I don't think it's needed because of the logic... the connection is not accessed concurrently by threads until IsValidated is set to true (i.e.: when we receive the first frame) and once it's set to true, it's considered immutable.